molecules/preprocess/preprocess.py

```diff
@@ -0,0 +1,218 @@
+from __future__ import print_function
+
+import os
+import random
+import sys
+import tempfile
+
+import apache_beam as beam
+import tensorflow as tf
+import tensorflow_transform.beam.impl as beam_impl
+
+from apache_beam.io import tfrecordio
+from apache_beam.options.pipeline_options import PipelineOptions
+from tensorflow_transform.beam.tft_beam_io import transform_fn_io
+from tensorflow_transform.coders import example_proto_coder
+from tensorflow_transform.saved import saved_transform_io
+from tensorflow_transform.tf_metadata import dataset_metadata
+from tensorflow_transform.tf_metadata import dataset_schema
+
+
+class ValidateInputData(beam.DoFn):
+  def __init__(self, input_metadata):
+    super(ValidateInputData, self).__init__()
+    self.schema_keys = set(input_metadata.schema.column_schemas.keys())
+
+  def process(self, elem):
+    if not isinstance(elem, dict):
+      raise ValueError(
+          'Element must be a dict(str, value). '
+          'Given: {} {}'.format(elem, type(elem)))
+    elem_keys = set(elem.keys())
+    if elem_keys != self.schema_keys:
+      raise ValueError(
+          "Element keys don't match the schema keys. "
+          'Given: {}; Schema: {}'.format(
+              list(elem_keys), list(self.schema_keys)))
+    yield elem
+
+
+def _make_train_or_eval_input_fn(
+    feature_spec, labels, file_names, batch_size, is_eval, shuffle=True):
+
+  def input_fn():
+    def decode(elem):
+      model_features = tf.parse_single_example(elem, features=feature_spec)
+      model_labels = tf.stack([model_features.pop(label) for label in labels])
+      return model_features, model_labels
+
+    dataset = tf.data.TFRecordDataset(file_names)
+    dataset = dataset.map(decode)
+    dataset = dataset.cache()
+    if not is_eval:
```

Comment:
maybe you can use the mode keys for this: https://www.tensorflow.org/api_docs/python/tf/estimator/ModeKeys